apps/condo/domains/notification/schema/MessageOrganizationBlackList.test.js

```diff
@@ -0,0 +1,186 @@
+                    await updateTestMessageOrganizationBlackList(anonymousClient, blackList.id, {
+                        description,
+                    })
+                })
+            })
+        })
+
+        describe('read', () => {
+            it('user cannot read MessageOrganizationBlackList', async () => {
+                const supportClient = await makeClientWithSupportUser()
+                const userClient = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                await createTestMessageOrganizationBlackList(supportClient)
+
+                await expectToThrowAccessDeniedErrorToObjects(async () => {
+                    await MessageOrganizationBlackList.getAll(userClient)
+                })
+            })
+
+            it('anonymous cannot read MessageOrganizationBlackList', async () => {
+                const supportClient = await makeClientWithSupportUser()
+                const anonymousClient = await makeClient()
+
+                await createTestMessageOrganizationBlackList(supportClient)
+
+                await expectToThrowAuthenticationErrorToObjects(async () => {
+                    await MessageOrganizationBlackList.getAll(anonymousClient)
+                })
+            })
+        })
+    })
+
+    describe('logic', () => {
+        it('dont send notification if organization added in MessageOrganizationBlackList', async () => {
+            const admin = await makeLoggedInAdminClient()
+            const userAttrs = {
+                name: faker.name.firstName(),
+                email: createTestEmail(),
+                phone: createTestPhone(),
+            }
+            const client = await makeClientWithRegisteredOrganization()
+
+            await createTestMessageOrganizationBlackList(admin, {
+                type: DIRTY_INVITE_NEW_EMPLOYEE_MESSAGE_TYPE,
+                organization: { connect: { id: client.organization.id } },
+            })
+
+            const [employee] = await inviteNewOrganizationEmployee(client, client.organization, userAttrs)
+
+            await waitFor(async () => {
+                const messageWhere = { user: { id: employee.user.id }, type: DIRTY_INVITE_NEW_EMPLOYEE_MESSAGE_TYPE }
+                const messages = await Message.getAll(admin, messageWhere)
+
+                expect(messages[0].status).toEqual(MESSAGE_ERROR_STATUS)
+                expect(messages[0].processingMeta.error).toEqual(MESSAGE_TYPE_IN_ORGANIZATION_BLACK_LIST)
+            })
+        })
+
+        it('dont send notification if all organizaitons added in MessageOrganizationBlackList', async () => {
+            const admin = await makeLoggedInAdminClient()
+            const userAttrs = {
+                name: faker.name.firstName(),
+                email: createTestEmail(),
+                phone: createTestPhone(),
+            }
+            const client = await makeClientWithRegisteredOrganization()
+
+            await createTestMessageOrganizationBlackList(admin, {
```

Comment:
don't you need to undo this after test? this would affect all organizations forever in other case.

Reply:
what if some day we want to block some kind of message to be sent to all organizations, and then, after some time, we would want to allow it for some of them? we would have to delete this rule and add it for all organizations except  allowed ones. maybe there should be whitelist too in order to make these more flexible?

Reply:
> don't you need to undo this after test? this would affect all organizations forever in other case.

deleted it in `beforeEach`

Reply:
> what if some day we want to block some kind of message to be sent to all organizations, and then, after some time, we would want to allow it for some of them? we would have to delete this rule and add it for all organizations except allowed ones. maybe there should be whitelist too in order to make these more flexible?

Added `MessageOrganizationWhiteList` for this case. The logic of the work is as follows: if a rule with empty organizations is set in the `MessageOrganizationBlackList` (block messages by type for all organizations), then adding a rule to the `MessageOrganizationWhiteList` will allow to send messages by type for a specific organization